### PR TITLE
GH-1034 allow languages to hook into the registration of server-side …

### PR DESCRIPTION
**Pull request by JanKoehnlein**
…commands

### CODE REVIEW

File: packages/languages/src/browser/language-client-contribution.ts [hunk 1]
```diff
@@ -38,7 +38,8 @@ export abstract class BaseLanguageClientContribution implements LanguageClientCo
     constructor(
         @inject(Workspace) protected readonly workspace: Workspace,
         @inject(Languages) protected readonly languages: Languages,
-        @inject(LanguageClientFactory) protected readonly languageClientFactory: LanguageClientFactory
+        @inject(LanguageClientFactory) protected readonly languageClientFactory: LanguageClientFactory,
+        @inject(Commands) protected readonly commands: Commands
```
Comment (akosyakov) [inline packages/languages/src/browser/language-client-contribution.ts]: you can use the property injection to avoid changing constructors of subclasses

Comment (JanKoehnlein) [reply packages/languages/src/browser/language-client-contribution.ts]: Aren't we using constructor injection by default all over the place?

Comment (akosyakov) [reply packages/languages/src/browser/language-client-contribution.ts]: I don't think we have such rule

**Comment by akosyakov** [comment]
How are you going to inject different instances of `Commands` after this change? With contextual bindings?

**Comment by JanKoehnlein** [comment]
@akosyakov By injecting the concrete classes in the constructor and passing them to super()

```
 constructor(
        @inject(Workspace) protected readonly workspace: Workspace,
        @inject(Languages) protected readonly languages: Languages,
	@inject(LanguageClientFactory) protected readonly languageClientFactory: LanguageClientFactory,
	@inject(GoCommands) protected readonly commands: GoCommands
    ) {
		super(workspace, languages, languageClientFactory, commands);
...
```

File: packages/languages/src/browser/language-client-contribution.ts (hunk 1 above)
Comment (svenefftinge) [reply packages/languages/src/browser/language-client-contribution.ts]: We at least used to have it and somehow gave it up last month without explicit discussion :)
But I like field injection, too, sometimes.

**Comment by svenefftinge** [comment]
Shouldn't we just let the BaseLanguageClientContribution directly implement `Commands` and move the code from `DefaultCommand` there? Then concrete classes could then easily override the method without that extra injection if they want to have different behavior. 
There is not really a lot of code in DefaultCommand.

**Comment by JanKoehnlein** [comment]
Test failures are unrelated

**Pull request merged**